syntax = "proto3";

package character.v1;

service CharacterService {
  rpc GetCharacterReward (RewardRequest) returns (RewardResponse) {}
}

message ClassificationSummary {
  string major_category = 1;
  string middle_category = 2;
  optional string minor_category = 3;
}

message RewardRequest {
  string source = 1; // enum: scan
  string user_id = 2; // UUID string
  string task_id = 3;
  ClassificationSummary classification = 4;
  repeated string situation_tags = 5;
  bool disposal_rules_present = 6;
  bool insufficiencies_present = 7;
}

message RewardResponse {
  bool received = 1;
  bool already_owned = 2;
  optional string name = 3;
  optional string dialog = 4;
  optional string match_reason = 5;
  optional string character_type = 6;
  optional string type = 7;
}
